branchpro/_incidence_number_plot.py

```diff
@@ -0,0 +1,81 @@
+#
+# IncidenceNumberPlot Class
+#
+# This file is part of BRANCHPRO
+# (https://github.com/SABS-R3-Epidemiology/branchpro.git) which is released
+# under the BSD 3-clause license. See accompanying LICENSE.md for copyright
+# notice and full license details.
+#
+
+import pandas as pd
+import plotly.graph_objs as go
+
+
+class IncidenceNumberPlot():
+    """IncidenceNumberPlot Class
+    Stores the main figure for the Dash app.
+    """
+    def __init__(self):
+        self._figure = go.Figure()
+
+    def add_data(self, df, time_key='Time', inc_key='Incidence Number'):
+        """
+        Supplies data to the figure which will be used for the bar plot.
+
+        Parameters
+        ----------
+        df
+            (pandas DataFrame) contains numbers of new cases by days.
+        time_key
+            x-axis label for the bar plot.
+        inc_key
+            y-axis label for the bar plot.
+        """
+        if type(df) != pd.DataFrame:
+            raise TypeError('Data needs to be a dataframe')
```

Comment:
I would use the variable name itself (df) instead of Data

Reply:
I was thinking to keep it consistent with the add_simulation type error

Reply:
I did change it though

Reply:
Okay I see, in that case I would change the message in add_simulation as well. In the traceback you'll always be told what function call you were in, so you should already know whether it is simulation or data, while it's helpful for the message to tell you exactly which argument to the method caused the problem.